File: backend/api/internal/port/inbound/events.go
package inbound

import (
	"context"

	"github.com/google/uuid"
	"github.com/mq/api/internal/domain/events"
)

type EventsService interface {
	List(ctx context.Context, filter events.ListFilter) ([]events.Event, error)
	ListPending(ctx context.Context) ([]events.Event, error)
	Search(ctx context.Context, query string, limit int) ([]events.Event, error)
	GetByID(ctx context.Context, id uuid.UUID) (*events.Event, error)
	GetBySlug(ctx context.Context, slug string) (*events.Event, error)
	AdminCreate(ctx context.Context, write events.EventWrite) (*events.Event, error)
	Submit(ctx context.Context, write events.EventWrite) (*events.Event, error)
	AdminUpdateContent(ctx context.Context, id uuid.UUID, write events.EventWrite) (*events.Event, error)
	AdminDelete(ctx context.Context, id uuid.UUID) error
	Review(ctx context.Context, id uuid.UUID, reviewerID uuid.UUID, status events.EventStatus, notes string) (*events.Event, error)
	Sync(ctx context.Context) (int, error)
}

File: apps/admin/src/lib/core/ports/events.ts
import type { Event, EventWrite, ReviewPayload, ReviewStatus, SyncEventsResult } from '$lib/core/domain/event';

export interface EventsPort {
  listPending(): Promise<Event[]>;
  getById(id: string): Promise<Event>;
  review(id: string, status: ReviewStatus, notes: string): Promise<Event>;
  sync(): Promise<SyncEventsResult>;
}

export type { Event, EventWrite, ReviewPayload, ReviewStatus, SyncEventsResult };
